From bcf2d788a8068e34519aa34b61ea200e43eae185 Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Wed, 16 Jan 2013 14:40:26 -0500
Subject: [PATCH 503/886] drm/radeon/dpm: remove broken dyn state remnants

commit 89c9bc565138ba7801e4ac1925ec9f013a8b4a57 upstream

Now that the proper fix has been implemented I can
remove the last remnants of the initial implementation.

Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/radeon/radeon.h    |    1 -
 drivers/gpu/drm/radeon/radeon_pm.c |   24 +++++-------------------
 2 files changed, 5 insertions(+), 20 deletions(-)

diff --git a/drivers/gpu/drm/radeon/radeon.h b/drivers/gpu/drm/radeon/radeon.h
index d9ab729..4ad02a3 100644
--- a/drivers/gpu/drm/radeon/radeon.h
+++ b/drivers/gpu/drm/radeon/radeon.h
@@ -1303,7 +1303,6 @@ struct radeon_dpm {
 	struct radeon_ps        *boot_ps;
 	/* default uvd power state */
 	struct radeon_ps        *uvd_ps;
-	struct radeon_ps        hw_ps;
 	enum radeon_pm_state_type state;
 	enum radeon_pm_state_type user_state;
 	u32                     platform_caps;
diff --git a/drivers/gpu/drm/radeon/radeon_pm.c b/drivers/gpu/drm/radeon/radeon_pm.c
index 4b80467..8ad4e04 100644
--- a/drivers/gpu/drm/radeon/radeon_pm.c
+++ b/drivers/gpu/drm/radeon/radeon_pm.c
@@ -684,17 +684,6 @@ restart_search:
 	return NULL;
 }
 
-static void radeon_dpm_update_requested_ps(struct radeon_device *rdev,
-					   struct radeon_ps *ps)
-{
-	/* copy the ps to the hw ps and point the requested ps
-	 * at the hw state in case the driver wants to modify
-	 * the state dynamically.
-	 */
-	rdev->pm.dpm.hw_ps = *ps;
-	rdev->pm.dpm.requested_ps = &rdev->pm.dpm.hw_ps;
-}
-
 static void radeon_dpm_change_power_state_locked(struct radeon_device *rdev)
 {
 	int i;
@@ -716,7 +705,7 @@ static void radeon_dpm_change_power_state_locked(struct radeon_device *rdev)
 
 	ps = radeon_dpm_pick_power_state(rdev, dpm_state);
 	if (ps)
-		radeon_dpm_update_requested_ps(rdev, ps);
+		rdev->pm.dpm.requested_ps = ps;
 	else
 		return;
 
@@ -767,11 +756,9 @@ static void radeon_dpm_change_power_state_locked(struct radeon_device *rdev)
 	down_write(&rdev->pm.mclk_lock);
 	mutex_lock(&rdev->ring_lock);
 
-	if (rdev->asic->dpm.pre_set_power_state) {
-		ret = radeon_dpm_pre_set_power_state(rdev);
-		if (ret)
-			goto done;
-	}
+	ret = radeon_dpm_pre_set_power_state(rdev);
+	if (ret)
+		goto done;
 
 	/* update display watermarks based on new power state */
 	radeon_bandwidth_update(rdev);
@@ -794,8 +781,7 @@ static void radeon_dpm_change_power_state_locked(struct radeon_device *rdev)
 	/* update current power state */
 	rdev->pm.dpm.current_ps = rdev->pm.dpm.requested_ps;
 
-	if (rdev->asic->dpm.post_set_power_state)
-		radeon_dpm_post_set_power_state(rdev);
+	radeon_dpm_post_set_power_state(rdev);
 
 done:
 	mutex_unlock(&rdev->ring_lock);
-- 
1.7.9.5

